java-manta-client/src/main/java/com/joyent/manta/util/MantaUtils.java

```diff
@@ -203,23 +204,27 @@ public static Boolean parseBooleanOrNull(final Object value) {
     }
 
     /**
-     * Format the path according to RFC3986.
+     * Format the manta path for inclusion in a URI path according to
+     * RFC3986.  Note that the encoding rules vary for each part of
+     * the url
      *
      * @param path the raw path string.
      * @return the URI formatted string with the exception of '/' which is special in manta.
-     * @throws UnsupportedEncodingException If UTF-8 is not supported on this system.
      */
-    public static String formatPath(final String path) throws UnsupportedEncodingException {
-        // first split the path by slashes.
-        final String[] elements = path.split("/");
-        final StringBuilder encodedPath = new StringBuilder();
-        for (final String string : elements) {
-            if (string.equals("")) {
-                continue;
-            }
-            encodedPath.append("/").append(URLEncoder.encode(string, "UTF-8"));
-        }
-        return encodedPath.toString();
+    public static String formatPath(final String path) {
+        final Encoder encoder = new Encoder(StandardCharsets.UTF_8);
```

Comment:
Ran unit tests plus `MantaClientPutIT,MantaClientIT,MantaDirectoryListingIteratorIT` with static instances and all seems well.  Will land with that change.